lib/reader/src/parser.rs

```diff
@@ -903,6 +931,68 @@ impl<'a> Parser<'a> {
         // Collect comments for the next instruction to be allocated.
         self.gather_comments(ctx.function.dfg.next_inst());
 
+        let mut encoding: Option<Encoding> = None;
+        let mut result_registers: Option<Vec<&'a str>> = None;
+
+        // encoding ::= "[" (encoding_literal | "-") , (result_registers | "-") "]"
+        if self.optional(Token::LBracket) {
+            let mut encoding_literal: Option<(&'a str, u16)> = None;
+
+            // encoding_literal ::= Identifier HexSequence
+            if let Some(Token::Identifier(recipe)) = self.token() {
+                self.consume();
+
+                if let Some(Token::HexSequence(bits_str)) = self.token() {
+                    self.consume();
```

Comment:
I would recommend adding `match_foo()` methods to avoid this error pyramid.

- `match_any_identifier()` is like `match_identifier()`, but allows any identifier text.
- `match_hex_sequence()` for `HexSequence`.